### ISSUE TITLE
Invalid cast from 'System.UInt64' to 'System.Byte[]'

### ISSUE THREAD

**Issue by kvpt** [borisdj/EFCore.BulkExtensions#539]
Hi,

Since version 3.4.7 I got an invalid cast exception related to the rowversion field when doing a BulkInsertOrUpdate.

I use the rowversion in combination with a ValueConverter, to have a ulong instead of a byte[], like this :

```csharp
entity.Property(e => e.RowVersion)
      .HasConversion(new NumberToBytesConverter<ulong>())
      .HasColumnType("timestamp")
      .IsRequired()
      .IsRowVersion()
      .IsConcurrencyToken();
````

This worked fine in version previous to 3.4.7.
I tried to exclude it (as it is auto generated) with PropertiesToExclude but it doesn't work.
The database is SqlServer.

Here is the complete stacktrace on version 5.1.2 :
```csharp
at Microsoft.Data.SqlClient.SqlBulkCopy.ConvertValue(Object value, _SqlMetaData metadata, Boolean isNull, Boolean& isSqlType, Boolean& coercedToDataFeed)
at Microsoft.Data.SqlClient.SqlBulkCopy.ReadWriteColumnValueAsync(Int32 col)
at Microsoft.Data.SqlClient.SqlBulkCopy.CopyColumnsAsync(Int32 col, TaskCompletionSource`1 source)
at Microsoft.Data.SqlClient.SqlBulkCopy.CopyRowsAsync(Int32 rowsSoFar, Int32 totalRows, CancellationToken cts, TaskCompletionSource`1 source)
at Microsoft.Data.SqlClient.SqlBulkCopy.CopyBatchesAsyncContinued(BulkCopySimpleResultSet internalResults, String updateBulkCommandText, CancellationToken cts, TaskCompletionSource`1 source)
at Microsoft.Data.SqlClient.SqlBulkCopy.CopyBatchesAsync(BulkCopySimpleResultSet internalResults, String updateBulkCommandText, CancellationToken cts, TaskCompletionSource`1 source)
at Microsoft.Data.SqlClient.SqlBulkCopy.WriteToServerInternalRestContinuedAsync(BulkCopySimpleResultSet internalResults, CancellationToken cts, TaskCompletionSource`1 source)
at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<InsertAsync>()
at EFCore.BulkExtensions.SQLAdapters.SQLServer.SqlOperationsServerAdapter.<MergeAsync>()
at EFCore.BulkExtensions.SqlBulkOperation.<MergeAsync>()
at EFCore.BulkExtensions.DbContextBulkTransaction.<ExecuteAsync>()
````

**Comment by borisdj** [comment]
Can you write entire code snipped to reproduce this.

**Comment by kvpt** [comment]
The only specific part necessary is the piece of code from above, the rest is rather standard.
Here the reproduction project : [EfCoreBulkIssue.zip](https://github.com/borisdj/EFCore.BulkExtensions/files/6390616/EfCoreBulkIssue.zip)

**Comment by borisdj** [comment]
Fixed with v5.1.3
